pennylane/resource/resource.py

```diff
@@ -125,6 +182,170 @@ def resources(self) -> Resources:
         """
 
 
+def add_in_series(r1: Resources, r2: Resources) -> Resources:
+    r"""Add two resources assuming the circuits are executed in series.
+
+    Args:
+        r1 (Resources): A Resources object to add.
+        r2 (Resources): A Resources object to add.
+
+    Returns:
+        Resources: The combined resources of r1 and r2.
+
+    .. details::
+
+        **Example**
+
+        >>> r1 = Resources(num_wires=2, num_gates=2, gate_types={'Hadamard': 1, 'CNOT':1}, gate_sizes={1: 1, 2: 1})
+        >>> r2 = Resources(num_wires=2, num_gates=2, gate_types={'RX': 1, 'CNOT':1}, gate_sizes={1: 1, 2: 1})
+        >>> print(r1 + r2)
+        wires: 2
+        gates: 4
+        depth: 0
+        shots: Shots(total=None)
+        gate_types:
+        {'Hadamard': 1, 'CNOT': 2, 'RX': 1}
+        gate_sizes:
+        {1: 2, 2: 2}
+    """
+
+    new_wires = max(r1.num_wires, r2.num_wires)
+    new_gates = r1.num_gates + r2.num_gates
+    new_gate_types = _combine_dict(r1.gate_types, r2.gate_types)
+    new_gate_sizes = _combine_dict(r1.gate_sizes, r2.gate_sizes)
+
+    return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes)
+
+
+def add_in_parallel(r1: Resources, r2: Resources) -> Resources:
+    r"""Add two resources assuming the circuits are executed in parallel.
+
+    Args:
+        r1 (Resources): A Resources object to add.
+        r2 (Resources): A Resources object to add.
+
+    Returns:
+        Resources: The combined resources of r1 and r2.
+
+    .. details::
+
+        **Example**
+
+        >>> r1 = Resources(num_wires=2, num_gates=2, gate_types={'Hadamard': 1, 'CNOT':1}, gate_sizes={1: 1, 2: 1})
+        >>> r2 = Resources(num_wires=2, num_gates=2, gate_types={'RX': 1, 'CNOT':1}, gate_sizes={1: 1, 2: 1})
+        >>> print(add_in_parallel(r1, r2))
+        wires: 4
+        gates: 4
+        depth: 0
+        shots: Shots(total=None)
+        gate_types:
+        {'Hadamard': 1, 'CNOT': 2, 'RX': 1}
+        gate_sizes:
+        {1: 2, 2: 2}
+    """
+
+    new_wires = r1.num_wires + r2.num_wires
+    new_gates = r1.num_gates + r2.num_gates
+    new_gate_types = _combine_dict(r1.gate_types, r2.gate_types)
+    new_gate_sizes = _combine_dict(r1.gate_sizes, r2.gate_sizes)
+
+    return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes)
```

Comment:
As long as we are generally supporting depth and shots in Resources, we should probably include them in the addition and multiplication methods.
For addition in parallel, (the new) depth would be the maximum.
For addition in series, (the new) depth would be the sum.
For multiplication in parallel, depth would go unchanged.
For multiplication in series, depth would be multiplied by the relevant scalar.
For Shots, there exists a dunder method for multiplication with a scalar, but as for addition I'm not 100% sure about the implementation.

Reply:
They're implemented now. For shots I defined addition as concatenating the shots vectors.